### DISCUSSION TITLE
[2024-10-14] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#11318] [Weekly Demo Day]
* #11161

[Screencast From 2024-09-24 14-13-51.webm](https://github.com/user-attachments/assets/ae9df4e2-dcc8-46aa-9977-32f470fd85fd)

* #11186 and #11198 :tada: 

[Screencast From 2024-10-14 16-30-35.webm](https://github.com/user-attachments/assets/b96051a2-49b5-4ec6-ac6e-a400b56e17d3)

* #11234
* #11287

![image](https://github.com/user-attachments/assets/a0209c20-113d-4fba-ad93-8bc31fdfb932)

* Near finish: #11271

[Screencast From 2024-10-14 13-51-24.webm](https://github.com/user-attachments/assets/372c6f1b-8281-4bdc-9f46-fa90c3e32764)

**Comment by JaroslavTulach** [comment]
Engine and language server GUI related changes:
- #11173
- #11218
- #11212
- Fixed Enso native parser crashes in the release - #11174 - @kazcw fixed it properly in `develop`
- #11250
- #11269
- [5% smaller caches on average](https://github.com/enso-org/enso/pull/11303#issuecomment-2407366513) - some even 28% smaller. #11303
- #11170
- #11200
- #11289 - @kazcw reported followed up issues for the parser & IDE
- #11204
- You can now `Panic.catch` in the IDE: #11167

**Comment by Frizi** [comment]
## 🍃 BAZEL progress

Already able to build the GUI frontend app after merging gui2 and dashboard:
<img width="891" alt="image" src="https://github.com/user-attachments/assets/0c5baabe-ae64-497a-9515-b0eaf75e161f">
 Remaining work:
- Adding support for environment variable inputs to the build. Currently the build relies on local .env file to function.
- Running the build in github action environment (planning to use [setup-bazel](https://github.com/marketplace/actions/setup-bazel) action)
- Defining all test jobs and running them on CI

**Comment by kazcw** [comment]
- #11147. A reported parser bug turned out to be a backend threading bug; I redesigned the compiler's parser bindings to be simpler and impossible to misuse.

Work in progress:
- #11240
